ambry-api/src/main/java/com.github.ambry/clustermap/ClusterMap.java

```diff
@@ -122,6 +122,15 @@
    */
   JSONObject getSnapshot();
 
+  /**
+   * Get new replica of certain partition that resides on given host.
+   * @param partitionIdStr the partition id string
+   * @param hostname the host on which replica should reside.
```

Comment:
`currentNode.hostname` and `currentNode.port` are passed in, why not use `DataNode` as argument?

Reply:
good idea. I will figure out a way to pass in `DataNodeId`

Reply:
sure, thanks for letting me know.